ReactAndroid/build.gradle

```diff
@@ -247,25 +247,11 @@ def getNdkBuildName() {
 }
 
 def findNdkBuildFullPath() {
```

Comment:
I think we should still keep all this, otherwise it may be breaking to some folks, even though some of these are being deprecated. Instead, maybe just add your new logic for detecting `project.android.ndkVersion`.

Then add TODO here to explain what should be deprecated.

Reply:
@fkgozali Just updated the pr, after debugging this yesterday I concluded that my new logic is not needed, we just need to make `android.ndkDirectory` the priority and `ANDROID_NDK` as the last priority.

But there is another problem that might be problematic to some folks. Because we specifically stated the ndkVersion here,
```
defaultConfig {
        applicationId "com.facebook.react.uiapp"
        applicationId "com.facebook.react.uiapp"
        minSdkVersion 19
        minSdkVersion 19
        ndkVersion "20.1.5948944"
}
```
Gradle will throw an exception saying that there is no version of NDK that matched the requested version. 

What do you think? if some folks run a different version than the one specified here they will get an error.

Reply:
Hmm, this is exactly what I was concerned with. I think to minimize surprise/breaking change atm, we should just make ANDROID_NDK the priority for now. If someone has it set up, we can assume that the machine has dedicated Android dev env set up, so let's trust the developer by using ANDROID_NDK.

Reply:
@fkgozali So the purpose of this pr is to fix the problem I have when trying to run RNTester with Gradle. The ReactAndroid Required me to use the older NDK 20.0.5594570 even when i already installed a specific NDK version on `ANDROID_NDK`. As far as I know, this is a bug from Gradle https://github.com/gradle/gradle/issues/12440#issuecomment-606188282. 

@friederbluemle suggest that there is three workaround 
1. Install the older NDK it is looking for (side-by-side): SDK manager 'ndk;20.0.5594570'
2. Manually specify the newer version in build.gradle. ( The one that I suggested on this pr ). 
3. Uninstall the NDK (if we don't need it): sdkmanager --uninstall 'ndk-bundle' 

This issue started happening when we upgraded to `com.android.tools.build:gradle:3.6.4` and seems this has not yet been fixed.

So I don't know what the best solution is, because even if I set the `ANDROID_NDK` correctly I still have to install `20.0.5594570` NDK because without this NDKReactAndroid won't build.

Reply:
> This issue started happening when we upgraded to com.android.tools.build:gradle:3.6.4 and seems this has not yet been fixed.

Side question: will version 4+ address this issue? Should we aim to upgrade to 4 before landing this PR?

Reply:
> Side question: will version 4+ address this issue? Should we aim to upgrade to 4 before landing this PR?

Haven't tried it yet, let me try.

Reply:
With Gradle 4.0 instead of `20.0.5594570 ` it asking for `21.0.6113669`. The problem still exist but the default NDK is changed to `21.0.6113669`. 

![Screenshot from 2020-09-24 01-10-21](https://user-images.githubusercontent.com/8868908/94052185-c4996b80-fe02-11ea-8cba-edb01b1ddad5.png)